hooks/init/31-upgrade-puppet.rb

```diff
@@ -52,7 +52,7 @@ def fail_and_exit(message)
   fail_and_exit 'Puppet already installed and upgraded. Skipping.' if File.exist?(PUPPET_UPGRADE_COMPLETE)
 
   katello = Kafo::Helpers.module_enabled?(@kafo, 'katello')
-  foreman_proxy_content = @kafo.param('foreman_proxy_plugin_pulp', 'pulpnode_enabled').value
+  foreman_proxy_content = Kafo::Helpers.module_enabled?(@kafo, 'foreman_proxy_content')
```

Comment:
I'm surprised we have a copy of that. I'd expect https://github.com/theforeman/kafo/blob/db5645b18cbce377ce089e3a898aee868e965070/lib/kafo/hook_context.rb#L70-L72 to work as well. Have you tried that?

Reply:
I have not, but I'd prefer to fix this BZ using our proven tooling, then tackle an issue to update usage across the board if thats fine?

Reply:
Fair enough for now.

Reply:
Filed a tracking issue -- http://projects.theforeman.org/issues/21324